packages/block-editor/src/components/global-styles/border-panel.js

```diff
@@ -7,16 +12,34 @@ import {
 	__experimentalIsDefinedBorder as isDefinedBorder,
 	__experimentalToolsPanel as ToolsPanel,
 	__experimentalToolsPanelItem as ToolsPanelItem,
+	__experimentalItemGroup as ItemGroup,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	__experimentalHeading as Heading,
+	__experimentalGrid as Grid,
+	__experimentalDropdownContentWrapper as DropdownContentWrapper,
+	BaseControl,
+	Dropdown,
+	Button,
+	FlexItem,
 } from '@wordpress/components';
 import { useCallback, useMemo } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import { shadow as shadowIcon, Icon, check } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import BorderRadiusControl from '../border-radius-control';
 import { useColorsPerOrigin } from './hooks';
 import { getValueFromVariable, TOOLSPANEL_DROPDOWNMENU_PROPS } from './utils';
+import { mergeOrigins } from '../../store/get-block-settings';
+import { setImmutably } from '../../utils/object';
+import { getBorderPanelLabel } from '../../hooks/border';
+
+function useHasShadowControl( settings ) {
+	return !! settings?.shadow;
+}
```

Comment:
It has to be mixed with border settings to `hide/show` the entire panel.
When shadow support is enabled and border support is disabled, entire panel disappears. Instead it should show shadow in the panel.